svm/src/transaction_processor.rs

```diff
@@ -228,9 +228,13 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         )));
 
         if programs_loaded_for_tx_batch.borrow().hit_max_limit {
+            const ERROR: TransactionError = TransactionError::ProgramCacheHitMaxLimit;
+            let loaded_transactions = vec![(Err(ERROR), None); sanitized_txs.len()];
+            let execution_results =
+                vec![TransactionExecutionResult::NotExecuted(ERROR); sanitized_txs.len()];
             return LoadAndExecuteSanitizedTransactionsOutput {
-                loaded_transactions: vec![],
-                execution_results: vec![],
+                loaded_transactions,
+                execution_results,
```

Comment:
@alessandrod can better decide if OK to return this error. But afaik, it is important to return same length results, cause call site often `zip` results with txs, or other pre-results. Is `zip-longest` an option?

Reply:
I don't think zip-longest is an option since Result doesn't have a default.
Also requires many more changes since we zip all over the place